### PR TITLE
WIP :enhance: kubelet-less support

### PR DESCRIPTION
**Pull request by jichenjc**
Fixes: #510

### CODE REVIEW

File: pkg/cgroup/resolve_container.go [hunk 1]
```diff
@@ -95,6 +95,7 @@ func GetContainerMetrics() (containerCPU, containerMem map[string]float64, nodeC
 }
 
 func GetAvailableKubeletMetrics() []string {
+	return []string{}
```
Comment (jichenjc) [inline pkg/cgroup/resolve_container.go:98]: this is for remove the kubelet metrics

after change: 
```
# curl http://localhost:9102/metrics | grep kubelet
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  120k    0  120k    0     0  23.5M      0 --:--:-- --:--:-- --:--:-- 23.5M
[root@jitest42 ~]#
```

File: pkg/cgroup/resolve_container.go [hunk 2]
```diff
@@ -116,22 +117,33 @@ func getContainerInfo(cGroupID, pid uint64, withCGroupID bool) (*ContainerInfo,
...
+				containerID = utils.SystemProcessName
+				// in addition to the system container ID, add also the system process name to the cache
+				if _, ok := containerIDToContainerInfo[containerID]; !ok {
+					containerIDToContainerInfo[containerID] = info
+				}
 			}
 		}
+	} else {
+
+		containerIDToContainerInfo[containerID] = &ContainerInfo{
```
Comment (jichenjc) [inline pkg/cgroup/resolve_container.go:141]: not able to get container name and pod info now ,so use N/A 

something like
```
[root@jitest42 ~]# curl http://localhost:9102/metrics | grep 'N/A'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0kepler_container_core_joules_total{command="",container_name="N/A",container_namespace="N/A",mode="dynamic",pod_name="N/A"} 475.6
kepler_container_core_joules_total{command="",container_name="N/A",container_namespace="N/A",mode="idle",pod_name="N/A"} 0
kepler_container_cpu_cpu_time_us{container_name="N/A",container_namespace="N/A",pod_name="N/A"} 8
kepler_container_dram_joules_total{command="",container_name="N/A",container_namespace="N/A",mode="dynamic",pod_name="N/A"} 7.5
kepler_container_dram_joules_total{command="",container_name="N/A",container_namespace="N/A",mode="idle",pod_name="N/A"} 0
kepler_container_gpu_joules_total{command="",container_name="N/A",container_namespace="N/A",mode="dynamic",pod_name="N/A"} 0
kepler_container_gpu_joules_total{command="",container_name="N/A",container_namespace="N/A",mode="idle",pod_name="N/A"} 0
kepler_container_joules_total{command="",container_name="N/A",container_namespace="N/A",mode="dynamic",pod_name="N/A"} 483.1
```

**Comment by jichenjc** [comment]
@rootfs a draft PR, can you check whether it's the desire of the issue? 
of course some hardcode existing, but want to check whether it's the right direction then can do more polish..

**Comment by rootfs** [comment]
@jichenjc Thanks for starting this thread. The behavior needs more consideration. My thinking is:
- If Kubelet is available but not accessible (e.g. kepler has wrong credential), we should let Kepler fail so that users can debug and get the right credential. 
- If not running in kubernetes environment and no containers are running, kepler then switches to the `kepler_process` metrics (see the WIP in #512)

**Comment by jichenjc** [comment]
>>>If not running in kubernetes environment and no containers are running, kepler then switches to the kepler_process metrics (see the WIP in https://github.com/sustainable-computing-io/kepler/issues/512)

@rootfs but what about has containers ,no k8s ? is PR #512 focusing pure process (so no container view?)

**Pull request closed without merging**